wasm/src/account/address.rs

```diff
@@ -165,6 +166,31 @@ impl Address {
     pub fn verify(&self, message: &[u8], signature: &Signature) -> bool {
         signature.verify(self, message)
     }
+
+    /// Get the address of a program based on the program ID.
+    ///
+    /// @param {string} program_id The program ID string.
+    /// @returns {Address} The address corresponding to the program ID.
+    #[wasm_bindgen(js_name = "fromProgramId")]
+    pub fn from_program_id(program_id: &str) -> Result<Self, String> {
+        // Convert the program ID string to a programId native type.
+        let program_id_native = ProgramIDNative::from_str(program_id).map_err(|e| e.to_string())?;
+
+        // Extracting the fields from the program name and network.
+        let name_field = program_id_native.name().to_field().map_err(|e| e.to_string())?;
+        let network_field = program_id_native.network().to_field().map_err(|e| e.to_string())?;
+
+        // Create a JS array of the fields.
+        let fields_array = js_array_from_fields!(&[name_field, network_field]);
+
+        // Initialize the Poseidon4 hasher.
+        let hasher = Poseidon4::new();
+
+        // Compute the group element corresponding to the program address.
+        let group = hasher.hash_to_group(fields_array).map_err(|e| e.to_string())?;
```

Comment:
Since at this point we're already in the context of `wasm` execution, we don't need use the SDK's external API in order to build this as it requires more roundtrips into JS land than necessary.

Once you have the name and network field this can be simplified to:
```suggestion
        // Compute the group element corresponding to the program address.
        let group = <CurrentNetwork as Network>::hash_to_group_psd4(&[name_field, network_field])?;
        Ok(Address::from(AddressNative::from(group)))
```